pkg/machine/applehv/machine.go

```diff
@@ -822,21 +823,18 @@ func getVMInfos() ([]*machine.ListResponse, error) {
 // setupStartHostNetworkingCmd generates the cmd that will be used to start the
 // host networking. Includes the ssh port, gvproxy pid file, gvproxy socket, and
 // a debug flag depending on the logrus log level
-func (m *MacMachine) setupStartHostNetworkingCmd(gvProxyBinary, forwardSock string, state machine.APIForwardingState) []string {
-	cmd := []string{gvProxyBinary}
-	// Add the ssh port
-	cmd = append(cmd, []string{"-ssh-port", fmt.Sprintf("%d", m.Port)}...)
-	// Add pid file
-	cmd = append(cmd, "-pid-file", m.GvProxyPid.GetPath())
-	// Add vfkit proxy listen
-	cmd = append(cmd, "-listen-vfkit", fmt.Sprintf("unixgram://%s", m.GvProxySock.GetPath()))
-	cmd, forwardSock, state = m.setupAPIForwarding(cmd)
-	if logrus.GetLevel() == logrus.DebugLevel {
-		cmd = append(cmd, "--debug")
-		fmt.Println(cmd)
-	}
-
-	return cmd
+func (m *MacMachine) setupStartHostNetworkingCmd() (gvproxy.Command, string, machine.APIForwardingState) {
+	cmd := gvproxy.NewCommand()
+	cmd.SSHPort = m.Port
+	cmd.PidFile = m.GvProxyPid.GetPath()
+	cmd.AddVfkitSocket(fmt.Sprintf("unixgram://%s", m.GvProxySock.GetPath()))
+	cmd.Debug = logrus.GetLevel() == logrus.DebugLevel
+
+	if cmd.Debug {
+		defer fmt.Println(cmd.ToCmdline())
+	}
```

Comment:
Why is this print defer?
Also while you are at it we should really juts use logrus.Debug()

Reply:
Ok yeah this is not obvious to a reader. My personal style is to only use defer for closing resources, i.e. close a file or lock.
To me this makes it harder to understand the logic because now the order of code and run is inverted. But this is juts personally preference so I wouldn't block on it.

Reply:
would doing as a debug be better?

Reply:
I'm not attached to the implementation. I don't mind adding some verbosity for the sake of readability